src/foraging_gui/bias_indicator.py

```diff
@@ -129,68 +144,99 @@ def calculate_bias(self,
-
-                # emit signal and flash current bias point if over
-                if abs(bias) > self.bias_threshold:
-                    self.log.info(f"Bias value calculated over a threshold of {self.bias_threshold}. Bias: {bias} "
-                                  f"Trial Count: {trial_count}")
-                    self.biasOver.emit(bias, trial_count)
-                    self._current_bias_point.setData(pos=[[time_point, bias]],
-                                                     pen=QColor('purple'),
-                                                     brush=QColor('purple'),
-                                                     size=9)
-
-                else:
-                    self._current_bias_point.setData(pos=[[time_point, bias]],
-                                                     pen=QColor('green'),
-                                                     brush=QColor('green'),
-                                                     size=9)
-
-            except ValueError as v:
-                acceptable_errors = [f'Cannot have number of splits n_splits={cv} greater than the number of samples:',
-                                     f'n_splits={cv} cannot be greater than the number of members in each class.',
-                                     'This solver needs samples of at least 2 classes in the data']
-                if any(x in str(v) for x in acceptable_errors):
-                    self.log.info(f"Can't calculate bias at trial count {trial_count} because {str(v).lower()}")
-                    self.biasError.emit(str(v), trial_count)
-                else:
-                    raise v
+                # add confidence intervals
+                upper = lr['df_beta'].loc['bias']['bootstrap_CI_upper'].values[0]
+                lower = lr['df_beta'].loc['bias']['bootstrap_CI_lower'].values[0]
+            elif len(unique) == 0:
+                # no choices, report bias confidence as (-inf, +inf)
+                bias = 0
+                upper = 1
+                lower = -1
+            elif unique[0] == 0:
+                # only left choices, report bias confidence as (-inf, 0)
+                bias = -1
+                upper = 0
+                lower = -1
+
+            elif unique[0] == 1:
+                # only right choices, report bias confidence as (0, +inf)
+                bias = 1
+                upper = 1
+                lower = 0
+
+            else:   # skip bias calculation if no conditions are met
+                return
+
+            # update
+            self._upper_scatter_item.setData(x=np.append(self._upper_scatter_item.xData, trial_num),
+                                             y=np.append(self._upper_scatter_item.yData, bias+upper))
+            self._lower_scatter_item.setData(x=np.append(self._lower_scatter_item.xData, trial_num),
+                                             y=np.append(self._lower_scatter_item.yData, bias+lower))
+
```

Comment:
@alexpiet I'm calculating the upper and lower ci coords with logic: 
upper_coord = bias + upper 
lower_coord = bias + lower (since lower is always negative)

I'm pretty sure that's correct but let me know if I'm missing an edge case or something

Reply:
No, the upper_coord is just 'upper'. and the lower_coord is just 'lower'. You don't need to add the 'bias' in

Reply:
Ohh okay great that is why sometimes they were off. Thanks for clarifying!